tests/util_test.py

```diff
@@ -55,3 +57,103 @@ def test_SExpr_format():
 #     assert unescape_string('\\u3042') == 'あ'
 #     assert unescape_string('\\xe3\\x81\\x82') == 'あ'
 #     assert unescape_string('\\N{HIRAGANA LETTER A}') == 'あ'
+
+@pytest.fixture
+def empty_file(tmpdir):
+    f = tmpdir.join('empty.txt')
+    return str(f)
+
+
+@pytest.fixture
+def nocomment_file(tmpdir):
+    f = tmpdir.join('nocomment.txt')
+    f.write('avm := *top*.')
+    return str(f)
+
+
+@pytest.fixture
+def utf8_file(tmpdir):
+    f = tmpdir.join('utf8.txt')
+    f.write('; coding: utf-8\n'
+            u'a := character & [ ORTH \"あ\" ].')
```

Comment:
I went back to the py.path docs and saw there's a [`write_text()`](https://py.readthedocs.io/en/latest/path.html#py._path.local.LocalPath.write_text) method which lets you specify the encoding. This (and all) should switch to use `write_text()` so you can write the file in the intended encoding.

Reply:
Also if you make one string `u'...'` you should make all of them that way so it's clear what the result of the implicit concatenation is.